api/encoding/call_test.go

```diff
@@ -52,15 +53,16 @@ func TestNilCall(t *testing.T) {
 func TestReadFromRequest(t *testing.T) {
 	ctx, icall := NewInboundCall(context.Background())
 	icall.ReadFromRequest(&transport.Request{
-		Service:         "service",
-		Transport:       "transport",
-		Caller:          "caller",
-		Encoding:        transport.Encoding("raw"),
-		Procedure:       "proc",
-		ShardKey:        "sk",
-		RoutingKey:      "rk",
-		RoutingDelegate: "rd",
-		CallerProcedure: "cp",
+		Service:           "service",
+		Transport:         "transport",
+		Caller:            "caller",
+		Encoding:          transport.Encoding("raw"),
+		Procedure:         "proc",
+		ShardKey:          "sk",
+		RoutingKey:        "rk",
+		RoutingDelegate:   "rd",
+		CallerProcedure:   "cp",
+		CallerPeerAddress: "foohost:1234",
```

Comment:
(1) The name doesn't align, `CallerPeerAddrPort`? 
(2) Do we need the port?
(3) Can we avoid the strings? https://pkg.go.dev/net/netip#AddrPort is a nice type.

Reply:
1) Done.
(2) Probably not but what's the harm ?
(3) I thought hard about this. As per the commit message, the intent is for logging/auditing, and using an netip.AddrPort limits us to TCP/UDP (v4 and v6) IP (maybe not a big deal), so the tradeoff is that things like non TCP/IP protocols don't fit into this model nicely.

Although in practice, in HTTP and TChannel, these are just passed as strings, and in GRPC, we are using peer.FromContext(ctx) which returns a https://pkg.go.dev/net#Addr. 

So I'm happy to parse it, but since the lower layers aren't doing that, I'm not sure if the upper layers should.

Reply:
>So I'm happy to parse it, but since the lower layers aren't doing that, I'm not sure if the upper layers should.

My thinking was the opposite - the upper layer should do the right thing from the start since changing them is harder. Internals we can fix/refactor later.

Reply:
@jquirke 
1. Did you pushed or changes? Seems like it's CallerPeerAddress everywhere.
3. My +1 for using https://pkg.go.dev/net/netip#AddrPort.
In addition, doc for public functions should have strong encourage to use 
https://pkg.go.dev/net/netip#AddrPort.IsValid with the result of call.

Reply:
1. Sorry yes, I didn't push. Fixed.
2. Done.